yb-voyager/cmd/importData.go

```diff
@@ -746,22 +761,35 @@ func importData(importFileTasks []*ImportFileTask, errorPolicy errorpolicy.Error
 			if err != nil {
 				utils.ErrExit("failed to restore sequences: %s", err)
 			}
-			displayImportedRowCountSnapshot(state, importFileTasks)
 		}
+		err = restoreGeneratedIdentityColumns(importTableList)
+		if err != nil {
+			utils.ErrExit("failed to restore generated columns: %s", err)
+		}
+		displayImportedRowCountSnapshot(state, importFileTasks)
 	}
 	fmt.Printf("\nImport data complete.\n")
+}
 
-	switch importerRole {
-	case TARGET_DB_IMPORTER_ROLE:
-		importDataCompletedEvent := createSnapshotImportCompletedEvent()
-		controlPlane.SnapshotImportCompleted(&importDataCompletedEvent)
-		packAndSendImportDataPayload(COMPLETE, "")
-	case SOURCE_REPLICA_DB_IMPORTER_ROLE:
-		packAndSendImportDataToSrcReplicaPayload(COMPLETE, "")
-	case SOURCE_DB_IMPORTER_ROLE:
-		packAndSendImportDataToSourcePayload(COMPLETE, "")
+func restoreGeneratedIdentityColumns(importTableList []sqlname.NameTuple) error {
```

Comment:
Are these operations (enableGeneratedAlwaysAsIdentityColumns, restoreGeneratedByDefaultAsIdentityColumns) idempotent? 
I.e. if these ALTERs are already run, and on restart of import-data, they are run again, that will not be a problem, right?

Reply:
Yes, these types of ALTERs are meant to be executed multiple times (to restore sequence values as well), so it isn't a problem.